src/plugins/summaryWidget/src/telemetry/SummaryWidgetMetadataProvider.js

```diff
@@ -0,0 +1,95 @@
+define([
+
+], function (
+
+) {
+
+    function SummaryWidgetMetadataProvider(openmct) {
+        this.openmct = openmct;
+    }
+
+    SummaryWidgetMetadataProvider.prototype.appliesTo = function (domainObject) {
+        return domainObject.type === 'summary-widget';
+    };
+
+    SummaryWidgetMetadataProvider.prototype.getDomains = function (domainObject) {
+        return this.openmct.time.getAllTimeSystems().map(function (ts, i) {
+            return {
+                key: ts.key,
+                name: 'UTC',
+                format: ts.timeFormat,
+                hints: {
+                    domain: i
+                }
+            };
+        });
+    };
+
+    SummaryWidgetMetadataProvider.prototype.getMetadata = function (domainObject) {
+        var enumerations = domainObject
+            .configuration
+            .ruleOrder
+            .map(function (ruleId, ruleIndex) {
+                return {
+                    string: domainObject.configuration.ruleConfigById[ruleId].label,
+                    value: ruleIndex
+                };
+            });
+
+        var metadata = {
+            // Generally safe assumption is that we have one domain per timeSystem.
+            values: this.getDomains().concat([
+                {
+                    name: 'state',
+                    key: 'state',
+                    source: 'ruleIndex',
+                    format: 'enum',
+                    enumerations: enumerations,
+                    hints: {
+                        range: 1
+                    }
+                },
+                {
+                    name: 'Rule Label',
+                    key: 'ruleLabel',
+                    format: 'string'
+                },
+                {
+                    name: 'Rule Name',
+                    key: 'ruleName',
+                    format: 'string'
+                },
+                {
+                    name: 'Message',
+                    key: 'message',
+                    format: 'string'
```

Comment:
A couple of points here, none of which need addressing immediately - 

* We will need to make sure we document the relationship between this metadata attribute and formatters themselves. We also need to make sure we clarify the role of formatters.
* Our formatters have also become 'interpreters' of data, because they will also return the data in the canonical javascript type appropriate for handling it if you use `.parse()`. So I wonder if 'formatter' is really the terminology we should be using.

Reply:
Yes-- formatter is a term I'm having some trouble with.  And their APIs are really not clearly documented.  Think there's a change coming.